src/pocketmine/Player.php

```diff
@@ -3237,6 +3249,26 @@ public function handleBookEdit(BookEditPacket $packet) : bool{
 		return true;
 	}
 
+	public function handleEmote(EmotePacket $packet) : bool{
+		$rawEmoteId = $packet->getEmoteId();
+		$emoteId = UUID::fromString($rawEmoteId);
+		$event = new PlayerEmoteEvent($this, $emoteId);
+		$event->call();
+		if ($event->isCancelled()) {
+			return true;
+		}
+
+		$newPacket = EmotePacket::create($this->getId(), $rawEmoteId, EmotePacket::FLAG_S2C);
+		$this->getLevelNonNull()->broadcastPacketToViewers($this, $newPacket);
```

Comment:
This should only be sent to the viewers of the player.

Reply:
In addition, this is not sufficient: if new players join, they won't receive this packet.

Reply:
Is it necessary to show the emote by delaying the packet transmission for other players who joined immediately after the emote started?

Reply:
No, that's my mistake. This PR is worse than I thought. It doesn't bother to broadcast `EmoteListPacket` _at all_.

Reply:
In this PR, I suggests to broadcast **only** `EmotePacket`

Reply:
Then why bother with `EmoteListPacket` at all? It's completely useless unless you're going to broadcast it.

Reply:
Further investigation into the purpose of `EmoteListPacket` is needed. If this PR works as intended, `EmoteListPacket` would appear to be entirely useless.

Reply:
I intended to be able to get the value for the future, but remove it if not needed